### PR TITLE
Full Transaction UX Overhaul

### PR DESCRIPTION
**Pull request by Romulus10**

### CODE REVIEW

**Comment by isaacadams** [comment]
> > 1. restored wallet
> > 2. created identity
> > 3. generated key
> > 4. published key
> > 5. received the following error
> > 
> > image
> 
> What does your browser console look like?

image

**Comment by Romulus10** [comment]
Are you running the most recent fennel-api?

File: src/views/pages/Identity/index.jsx
Comment (isaacadams) [inline src/views/pages/Identity/index.jsx:8]: please ensure this line (and the other related lines) remain and don't get blown out with the merge. This is the logic that conditionally renders this based on whether an account is loaded.

**Comment by isaacadams** [comment]
> Are you running the most recent fennel-api?

yes, I pulled everything down before running.

**Comment by Romulus10** [comment]
> > Are you running the most recent fennel-api?
> 
> 
> 
> yes, I pulled everything down before running.

Through Docker?

**Comment by isaacadams** [comment]
> > > Are you running the most recent fennel-api?
> > 
> > 
> > yes, I pulled everything down before running.
> 
> Through Docker?

Yes. I ran `git pull` and then `docker-compose up -d --build` and ensured that all the containers logs were in a ready state -- I am still getting this issue.

**Comment by Romulus10** [comment]
Under fennel-api what does your nginx/nginx.conf look like?

**Comment by isaacadams** [comment]
> 

I haven't changed any files, if that is what you are asking. It is the same as whats in source control on the master branch.

**Comment by Romulus10** [comment]
I didn't figure you'd changed anything, I just wanted to exhaust any configuration conflicts to cross those off early.
We had a similar problem during review where the selected nginx container behaved differently than expected while echoing the selected CORS headers back to the client. I worked with the evaluator to resolve that problem and they were able to successfully connect to the API, so this seems to be a totally new issue.

**Comment by Romulus10** [comment]
That time it was that it wasn't echoing back the necessary header at all, now it's doing it too many times.

File: src/services/Node/index.js
Comment (isaacadams) [inline src/services/Node/index.js:26]: great use of the behavior subject and its associated methods for displaying the fee for any given tool. The logic here is sound: the user will only ever be submitting one transaction at a time, and therefore, we only need to keep track of one fee state at a time for any given kind of transaction.

File: src/services/ContactsManager.service.js
Comment (isaacadams) [inline src/services/ContactsManager.service.js:54]: not critical for now, but I believe promises require a default `catch` implementation. If there is none, it will complain. Again, not critical, but good to keep in mind for future housekeeping.

File: src/views/pages/Identity/index.jsx
Comment (isaacadams) [inline src/views/pages/Identity/index.jsx:18]: I think we should keep the IdentityControls component as a sub-component to this page. It will help reduce the complexity.

**Review by isaacadams — CHANGES_REQUESTED** [review]
the only thing I see that needs to be fixed before approval is in Identity/index.jsx. The changes, as is, will blow away the `useAccount` implementation. Please make sure that remains in this merge. We should also keep the `<IdentityControls />` component as that helps reduce the complexity.

File: src/services/ContactsManager.service.js
Comment (Romulus10) [reply src/services/ContactsManager.service.js:54]: This is a symptom of a mismatch in design pattern throughout the app. We'll review.

**Comment by Romulus10** [comment]
> the only thing I see that needs to be fixed before approval is in Identity/index.jsx. The changes, as is, will blow away the `useAccount` implementation. Please make sure that remains in this merge. 

It's not a priority now, but we need to add an issue to make sure that useAccount is integrated with the rest of the app.

> We should also keep the `<IdentityControls />` component as that helps reduce the complexity.

I added a good bit of functionality to that page, so it's likely going to need a total rewrite. It does need streamlining. Let's also mark this as an issue.

**Review by isaacadams — APPROVED** [review]

**Pull request merged**
